### DISCUSSION TITLE
Split router to multiple files?

### DISCUSSION THREAD

**Discussion by jhjacobs81** [salvo-rs/salvo#242] [General]
Is it possible to split the router into seperate files?

suppose i want to build a social media site of some sorts, which includes a forum, a wiki, a gallery, a marketplace, etc etc. After a while main.rs gets rather big, so i would like to split it up into a module called router, which then has forum.rs, wiki.rs, gallery.rs which then includes the routes for their respective modules. 

Is this possible?

**Comment by chrislearn** [comment]
Of course you can and you should. A example is here: 
https://github.com/driftluo/myblog/blob/f2ac8875896d8f1217658f84fc6f0a9a2a990d2c/src/bin/blog.rs#L39-L59

In fact, router's structure is a tree, you can split any part of it to a new module.

```rust
Router::new().push(user::root_router())
    .push(wiki::root_router())
```
